j5/boards/base.py

```diff
@@ -27,6 +27,7 @@ def __str__(self) -> str:
     def __new__(cls, *args, **kwargs):
         """Ensure any instantiated board is correctly setup."""
         instance = super().__new__(cls)
+        instance._board_setup = False
```

Comment:
Not a huge fan of this monkey patching, but as this needs to be an instance attribute I can't see a way around it. As it is an internal attribute not meant to be used by consumers it makes me feel slightly better about this. Happy to discuss alternative options.

Reply:
What's wrong with `board in BOARDS`?

Reply:
Specifically:

```python
def _setup(self):
  if self not in Board.BOARDS:
    Board.BOARDS.append(self)
  else:
    raise Exception(f"Setup has already been called for board {self.serial}")
```

Reply:
Render moot by later discussion, but as mentioned in a previous comment this was about conveying intent clearly.